zokrates_core/src/semantics.rs

```diff
@@ -1176,29 +1153,6 @@ impl<'ast, T: Field> Checker<'ast, T> {
 
                     match self.check_statement(stat, module_id, &state.types) {
                         Ok(statement) => {
-                            if let TypedStatement::Return(e) = &statement {
```

Comment:
Is this being checked somewhere else? The type of the return value must match the return type of the signature

Reply:
yes, when the return statement is checked: https://github.com/Zokrates/ZoKrates/blob/deprecate-multi-returns/zokrates_core/src/semantics.rs#L1724

but I wonder about this other case when the return expression is None: https://github.com/Zokrates/ZoKrates/blob/deprecate-multi-returns/zokrates_core/src/semantics.rs#L1709

might need to add a check there also